### PR TITLE
Enable insfactor.el for cider

### PR DESCRIPTION
**Pull request by deobald**
Hello Mr. John Hume,

I write to you today to request that you joyfully receive this hackity hack which enables insfactor.el on cider-based configurations. In time, I expect this will become unnecessary as the elisp government settles on a final name for this goddamn thing. In the meantime, I would very much like to use your tool. I think it is nifty.

Warm thanks and regards,
Steven Deobald The First.

### CODE REVIEW

**Comment by duelinmarkers** [comment]
I extend my warmest thanks for your kind and generous hackity hack. I hope to test and merge it after the dinner hour this evening. If I should be delayed, please do not think me ungrateful.

I do hope Bethany and the children are getting on well!

**Comment by duelinmarkers** [comment]
Hmm ... I get `save-current-buffer: Wrong type argument: stringp, nil` when I `M-x insfactor-index-project`. If I eval just the first interesting line of that fun, I get this backtrace, for which I apologize in advance.

```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  set-buffer(nil)
  (save-current-buffer (set-buffer (nrepl-current-connection-buffer)) nrepl-session)
  nrepl-current-session()
  (or session (nrepl-current-session))
  (list "op" "eval" "session" (or session (nrepl-current-session)) "code" input)
  (append (if ns (list "ns" ns)) (list "op" "eval" "session" (or session (nrepl-current-session)) "code" input))
  nrepl-eval-request("(try\n                        (require 'duelinmarkers.insfactor.project)\n                        (catch Exception e (println e)))" nil nil)
  (nrepl-send-request (nrepl-eval-request input ns session) callback)
  (let ((ns (if (string-match "[[:space:]]*(ns\\([[:space:]]*$\\|[[:space:]]+\\)" input) "user" ns))) (nrepl-send-request (nrepl-eval-request input ns session) callback))
  nrepl-send-string("(try\n                        (require 'duelinmarkers.insfactor.project)\n                        (catch Exception e (println e)))" (lambda (&rest --cl-rest--) (apply (quote (lambda (G15704 G15705 G15706 G15707 G15708 G15709 response) (let ((value (rest ...)) (ns (rest ...)) (out (rest ...)) (err (rest ...)) (status (rest ...)) (id (rest ...)) (ex (rest ...)) (root-ex (rest ...)) (session (rest ...))) (cond (value (save-current-buffer ... ...) (if ... ...)) (out (if ... ...)) (err (if ... ...)) (status (if ... ...) (if ... ...) (if ... ...) (if ... ...) (if ... ...)))))) (quote --eval-error-handler--) (quote --done-handler--) (quote --stderr-handler--) (quote --stdout-handler--) (quote --value-handler--) (quote --buffer--) --cl-rest--)))
  eval((nrepl-send-string "(try\n                        (require 'duelinmarkers.insfactor.project)\n                        (catch Exception e (println e)))" (nrepl-make-response-handler (nrepl-current-repl-buffer) (lambda (buffer v) (message "%s" v)) (lambda (buffer v) (message "%s" v)) (lambda (buffer v) (message "%s" v)) (lambda (buffer v) (message "%s" v)))) nil)
  eval-last-sexp-1(nil)
  ad-Orig-eval-last-sexp(nil)
  #[(&rest _it) "\302!\211\207" [eval-last-sexp-arg-internal ad-return-value ad-Orig-eval-last-sexp] 2]()
  funcall(#[(&rest _it) "\302!\211\207" [eval-last-sexp-arg-internal ad-return-value ad-Orig-eval-last-sexp] 2])
  nrepl-eval-sexp-fu-flash-doit-simple(#[(&rest _it) "\302!\211\207" [eval-last-sexp-arg-internal ad-return-value ad-Orig-eval-last-sexp] 2] (closure (t) (&rest args) (apply (quote nesf-hl-highlight-bounds) (quote (1743 . 2320)) (quote nrepl-eval-sexp-fu-flash) (quote #<buffer insfactor.el>) args)) (closure (t) (&rest args) (apply (quote nesf-hl-unhighlight-bounds) (quote (1743 . 2320)) (quote #<buffer insfactor.el>) args)))
  funcall(nrepl-eval-sexp-fu-flash-doit-simple #[(&rest _it) "\302!\211\207" [eval-last-sexp-arg-internal ad-return-value ad-Orig-eval-last-sexp] 2] (closure (t) (&rest args) (apply (quote nesf-hl-highlight-bounds) (quote (1743 . 2320)) (quote nrepl-eval-sexp-fu-flash) (quote #<buffer insfactor.el>) args)) (closure (t) (&rest args) (apply (quote nesf-hl-unhighlight-bounds) (quote (1743 . 2320)) (quote #<buffer insfactor.el>) args)))
  nrepl-eval-sexp-fu-flash-doit(#[(&rest _it) "\302!\211\207" [eval-last-sexp-arg-internal ad-return-value ad-Orig-eval-last-sexp] 2] (closure (t) (&rest args) (apply (quote nesf-hl-highlight-bounds) (quote (1743 . 2320)) (quote nrepl-eval-sexp-fu-flash) (quote #<buffer insfactor.el>) args)) (closure (t) (&rest args) (apply (quote nesf-hl-unhighlight-bounds) (quote (1743 . 2320)) (quote #<buffer insfactor.el>) args)))
  (prog1 (nrepl-eval-sexp-fu-flash-doit do-it-thunk hi unhi) (setq ret t))
  (unwind-protect (prog1 (nrepl-eval-sexp-fu-flash-doit do-it-thunk hi unhi) (setq ret t)) (if ret nil (funcall eflash)))
  (let (ret) (unwind-protect (prog1 (nrepl-eval-sexp-fu-flash-doit do-it-thunk hi unhi) (setq ret t)) (if ret nil (funcall eflash))))
  nesf-flash-doit(#[(&rest _it) "\302!\211\207" [eval-last-sexp-arg-internal ad-return-value ad-Orig-eval-last-sexp] 2] (closure (t) (&rest args) (apply (quote nesf-hl-highlight-bounds) (quote (1743 . 2320)) (quote nrepl-eval-sexp-fu-flash) (quote #<buffer insfactor.el>) args)) (closure (t) (&rest args) (apply (quote nesf-hl-unhighlight-bounds) (quote (1743 . 2320)) (quote #<buffer insfactor.el>) args)) (closure (t) (&rest args) (apply (quote nesf-flash-error-bounds) (quote (1743 . 2320)) (quote #<buffer insfactor.el>) (quote nrepl-eval-sexp-fu-flash-error) args)))
  eval-last-sexp(nil)
  call-interactively(eval-last-sexp nil nil)
```

**Comment by duelinmarkers** [comment]
Hold on. I did something real stupid.

**Comment by duelinmarkers** [comment]
It works. Hoping those hardasses over at melpa will be ok with it.

**Comment by duelinmarkers** [comment]
The big ugly backtrace above is what you get when you don't have an nrepl connection because you forgot to jack-in or you jacked into insfactor itself while having left a syntax error in insfactor.clj, so I advise not to do those things.

**Comment by deobald** [comment]
Curious! Yeah, sorry... Because the fn signatures were the same I just
swapped them out one-for-one. I didn't actually go back and test on nrepl
because it seemed like it should work as it always had.

I'll get an nrepl installation going somewhere and see what's up.

Apologies!

**Comment by deobald** [comment]
Hot.

...melpa, the new debian? :P

File: insfactor.el [hunk 1]
```diff
@@ -33,7 +33,23 @@
...
 
-(require 'nrepl)
+(defvar prefix)
+
+(when (featurep 'nrepl)
+  (require 'nrepl)
+  (setq prefix "nrepl-"))
+
+(when (featurep 'cider-autoloads)
+  (require 'cider-autoloads)
```
Comment (purcell) [inline insfactor.el:43]: This is wrong -- don't use cider-autoloads. They may not even be there if cider was installed manually. Instead, you probably want to try `(require 'cider nil t)`, and then look for `(featurep 'cider)`.

Comment (duelinmarkers) [inline insfactor.el:43]: Thanks. How does this look? a6800ded3f92f5ace1276cb2e6ad262c5fe3753e

Comment (purcell) [inline insfactor.el:43]: Yeah, that looks much better!

Comment (purcell) [inline insfactor.el:43]: Just sent you a couple of further changes in https://github.com/duelinmarkers/insfactor.el/pull/5 :-)

**Pull request merged**